### ISSUE TITLE
bug: java.lang.Throwable: Control-flow exceptions (e.g. this class com.intellij.openapi.progress.CeProcessCanceledException) shoul...

### ISSUE THREAD

**Issue by mostlydev** [sourcegraph/jetbrains#2467]
### IDE Information

JetBrains Rider 2024.1.4
Build #RD-241.18034.76, built on June 23, 2024
Subscription is active until November 7, 2024.
Runtime version: 17.0.11+1-b1207.24 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Windows 11.0
.NET Core v8.0.1 x64 (Server GC)
GC: G1 Young Generation, G1 Old Generation
Memory: 4056M
Cores: 8
Registry:
  ide.experimental.ui=true
  ide.new.project.model.index.case.sensitivity=true
Non-Bundled Plugins:
  com.github.continuedev.continueintellijextension (0.0.75)
  com.intellij.resharper.StructuredLogging (2024.1.4.331)
  wang.zhanwei.clangformat (1.4.0-RELEASE)
  Key Promoter X (2024.2.0)
  fr.socolin.application-insights-debug-log-viewer (2024.1.2)
  String Manipulation (9.14.1)
  com.supermaven.intellij (1.42)
  com.jetbrains.gerryPurpleTheme (2024.1.1010)
  com.intellij.plugin.adernov.powershell (2.6.1)
  quickturnstudio.sdk.language.qss.QtStyleSheetsEditor (0.9.4)
  com.intellij.javafx (1.0.4)
  net.sjrx.intellij.plugins.systemdunitfiles (223.241016.284)
  ru.adelf.idea.dotenv (2024.2.1)
  com.sourcegraph.jetbrains (7.0.13)

### Bug Description

Not sure what's happening here, but the whole plugin crashes, and the error suggests that it's because of a syntax error in an open file in a different window.  This error comes up in parallel and the plugin fails to display completely

### Additional context

Stacktrace:
```text
java.lang.Throwable: Control-flow exceptions (e.g. this class com.intellij.openapi.progress.CeProcessCanceledException) should never be logged. Instead, these should have been rethrown if caught.
	at com.intellij.openapi.diagnostic.Logger.ensureNotControlFlow(Logger.java:513)
	at com.intellij.idea.IdeaLogger.doLogError(IdeaLogger.java:154)
	at com.intellij.idea.IdeaLogger.error(IdeaLogger.java:145)
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:436)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService.lambda$submit$1(PassExecutorService.java:336)
	at com.intellij.concurrency.JobLauncherImpl$VoidForkJoinTask$1.exec(JobLauncherImpl.java:200)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
Caused by: com.intellij.openapi.progress.CeProcessCanceledException: com.intellij.openapi.progress.IndicatorCancellationException
	at com.intellij.serviceContainer.ComponentManagerImplKt$runBlockingInitialization$1.invoke(ComponentManagerImpl.kt:1639)
	at com.intellij.serviceContainer.ComponentManagerImplKt$runBlockingInitialization$1.invoke(ComponentManagerImpl.kt:1627)
	at com.intellij.openapi.progress.ContextKt.prepareIndicatorThreadContext$lambda$5(context.kt:107)
	at com.intellij.openapi.progress.impl.CoreProgressManager.silenceGlobalIndicator(CoreProgressManager.java:1022)
	at com.intellij.openapi.progress.ContextKt.prepareIndicatorThreadContext(context.kt:105)
	at com.intellij.openapi.progress.ContextKt.prepareThreadContext(context.kt:75)
	at com.intellij.serviceContainer.ComponentManagerImplKt.runBlockingInitialization(ComponentManagerImpl.kt:1627)
	at com.intellij.serviceContainer.ComponentManagerImplKt.getOrCreateInstanceBlocking(ComponentManagerImpl.kt:1547)
	at com.intellij.serviceContainer.ComponentManagerImpl.doGetService(ComponentManagerImpl.kt:754)
	at com.intellij.serviceContainer.ComponentManagerImpl.getService(ComponentManagerImpl.kt:698)
	at com.intellij.openapi.updateSettings.impl.UpdateChecker.updateAndShowResult(UpdateChecker.kt:932)
	at com.intellij.openapi.updateSettings.impl.UpdateChecker.checkForUpdate(UpdateChecker.kt:577)
	at com.intellij.diagnostic.DefaultIdeaErrorLogger.canHandle(DefaultIdeaErrorLogger.java:30)
	at com.intellij.diagnostic.DialogAppender.queueAppend(DialogAppender.kt:85)
	at com.intellij.diagnostic.DialogAppender.publish(DialogAppender.kt:57)
	at java.logging/java.util.logging.Logger.log(Logger.java:980)
	at java.logging/java.util.logging.Logger.doLog(Logger.java:1007)
	at java.logging/java.util.logging.Logger.log(Logger.java:1118)
	at com.intellij.idea.IdeaLogger.doLogError(IdeaLogger.java:169)
	at com.intellij.idea.IdeaLogger.error(IdeaLogger.java:145)
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:436)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$3(PassExecutorService.java:442)
	at com.intellij.openapi.application.impl.RwLockHolder.tryRunReadAction(RwLockHolder.kt:310)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:958)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$4(PassExecutorService.java:403)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$13(CoreProgressManager.java:660)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:735)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:691)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:659)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:79)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.doRun(PassExecutorService.java:402)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$run$0(PassExecutorService.java:378)
	at com.intellij.openapi.fileTypes.impl.FileTypeManagerImpl.cacheFileTypesInside(FileTypeManagerImpl.java:795)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$run$1(PassExecutorService.java:378)
	at com.intellij.openapi.application.impl.ReadMostlyRWLock.executeByImpatientReader(ReadMostlyRWLock.java:199)
	at com.intellij.openapi.application.impl.RwLockHolder.executeByImpatientReader(RwLockHolder.kt:491)
	at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:182)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.run(PassExecutorService.java:376)
	at com.intellij.concurrency.JobLauncherImpl$VoidForkJoinTask$1.exec(JobLauncherImpl.java:190)
	... 5 more
Caused by: com.intellij.openapi.progress.IndicatorCancellationException
	at com.intellij.openapi.progress.ContextKt$cancelWithIndicator$1.invokeSuspend(context.kt:133)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
	at kotlinx.coroutines.internal.LimitedDispatcher$Worker.run(LimitedDispatcher.kt:115)
	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:103)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:584)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:793)
	at kotlinx.c...

**Comment by PriNova** [comment]
Thank you for reporting the issue. Duplicate of #2316
